### ISSUE TITLE
Should we gitignore `package-lock.json`?

### ISSUE THREAD

**Issue by LeaVerou** [color-js/color.js#623]
For packages that are dependencies of other packages (such as Color.js), I'm not sure there is much value in committing `package-lock.json`. Should we just remove it from the repo?

cc @lloydk @facelessuser @MysteryBlokHed @jgerigmeyer

**Comment by lloydk** [comment]
I'd prefer to keep it because it ensures that developers are using the same versions of the development dependencies.

What are the downsides?

**Comment by facelessuser** [comment]
Yeah, if there is no adverse effect, I don't see a reason to not include it.

**Comment by LeaVerou** [comment]
If we want to restrict dependencies to specific versions we could always tighten the requirements in `package.json`. It seems to defeat the purpose to have ranges there, then enforce specific versions via `package-lock.json` (and can be wasteful as it creates more redundancy).

Some more context here: https://dev.to/gajus/stop-using-package-lock-json-or-yarn-lock-3ddi

**Comment by kleinfreund** [comment]
While you can control the versions of directly listed dependencies, the lock file allows you to also lock the versions of indirect/transient dependencies. It's not uncommon that an update in such a dependency causes issues in a project which is then harder to reproduce and also to avoid without resorting to overriding such faulty versions. With a lock file, this is much more transparent and safe. 

On the other hand, lock files tend to, over time, hold on to long outdated indirect dependency versions making it sometimes (though not commonly) to regenerate the lock file. However, that's a smaller problem than those lock files fix in my opinion.

**Comment by 43081j** [comment]
the various blog posts/opinions around not having a lockfile are pretty much all pointing out that the dependencies you install when developing a _library_ are not the same as the ones consumers will pull down (since consumers of a library don't use the library's lock file).

this is true, but meanwhile it is extremely useful that all contributors have the same dev setup

you could disable usage of the lockfile in CI so you're still testing against what a user would see. but that has its own downsides too - since contributors will have a different dependency tree to CI

given whatever you decide has a downside, i'd just flip a coin to be honest 😅

**Comment by facelessuser** [comment]
As far as I'm aware, all of Color.js's dependencies are dev dependencies, so I'm not sure the user issue is even coming into the equation. Color.js has no dependencies for user use as far as I know.

**Comment by 43081j** [comment]
very good point

in that case, we should keep it for sake of having better dx for contributors imo

**Comment by MysteryBlokHed** [comment]
I agree that there's not really any harm in keeping it around.

**Comment by facelessuser** [comment]
I've seen no real, tangible issues brought up. I would vote to close this.
